includes/import/abstract-wc-product-importer.php

```diff
@@ -0,0 +1,941 @@
+				if ( ! empty( $attribute['id'] ) ) {
+					$attribute_id   = absint( $attribute['id'] );
+					$attribute_name = wc_attribute_taxonomy_name_by_id( $attribute_id );
+				} elseif ( ! empty( $attribute['name'] ) ) {
+					$attribute_name = wc_clean( $attribute['name'] );
+				}
+
+				if ( ! $attribute_id && ! $attribute_name ) {
+					continue;
+				}
+
+				if ( $attribute_id ) {
+
+					if ( isset( $attribute['options'] ) ) {
+						$options = $attribute['options'];
+
+						if ( ! is_array( $attribute['options'] ) ) {
+							// Text based attributes - Posted values are term names.
+							$options = explode( WC_DELIMITER, $options );
+						}
+
+						$values = array_map( 'wc_sanitize_term_text_based', $options );
+						$values = array_filter( $values, 'strlen' );
+					} else {
+						$values = array();
+					}
+
+					if ( ! empty( $values ) ) {
+						// Add attribute to array, but don't set values.
+						$attribute_object = new WC_Product_Attribute();
+						$attribute_object->set_id( $attribute_id );
+						$attribute_object->set_name( $attribute_name );
+						$attribute_object->set_options( $values );
+						$attribute_object->set_position( isset( $attribute['position'] ) ? (string) absint( $attribute['position'] ) : '0' );
+						$attribute_object->set_visible( ( isset( $attribute['visible'] ) && $attribute['visible'] ) ? 1 : 0 );
+						$attribute_object->set_variation( ( isset( $attribute['variation'] ) && $attribute['variation'] ) ? 1 : 0 );
+						$attributes[] = $attribute_object;
+					}
+				} elseif ( isset( $attribute['options'] ) ) {
+					// Custom attribute - Add attribute to array and set the values.
+					if ( is_array( $attribute['options'] ) ) {
+						$values = $attribute['options'];
+					} else {
+						$values = explode( WC_DELIMITER, $attribute['options'] );
+					}
+					$attribute_object = new WC_Product_Attribute();
+					$attribute_object->set_name( $attribute_name );
+					$attribute_object->set_options( $values );
+					$attribute_object->set_position( isset( $attribute['position'] ) ? (string) absint( $attribute['position'] ) : '0' );
+					$attribute_object->set_visible( ( isset( $attribute['visible'] ) && $attribute['visible'] ) ? 1 : 0 );
+					$attribute_object->set_variation( ( isset( $attribute['variation'] ) && $attribute['variation'] ) ? 1 : 0 );
+					$attributes[] = $attribute_object;
+				}
+			}
+			$product->set_attributes( $attributes );
+		}
+
+		// Sales and prices.
+		if ( in_array( $product->get_type(), array( 'variable', 'grouped' ), true ) ) {
+			$product->set_regular_price( '' );
+			$product->set_sale_price( '' );
+			$product->set_date_on_sale_to( '' );
+			$product->set_date_on_sale_from( '' );
+			$product->set_price( '' );
+		} else {
+			// Regular Price.
+			if ( isset( $data['regular_price'] ) ) {
+				$product->set_regular_price( $data['regular_price'] );
+			}
+
+			// Sale Price.
+			if ( isset( $data['sale_price'] ) ) {
```

Comment:
There is actually no `sale_price` field in [the schema](https://github.com/woocommerce/woocommerce/wiki/Product-CSV-Import-Schema). There is a `regular_price` and a `price` which "may set sale or regular price depending on other columns". You'll need a little logic to figure out which field to set, depending on the whether `regular_price` is set and the value of `price`.